recipe/verify-version.cmake

```
@@ -0,0 +1,12 @@
+set(CCCL_VERSION_FILE $ENV{CCCL_VERSION_FILE})
+
+if(NOT CCCL_VERSION_FILE)
+    message(FATAL_ERROR "CCCL_VERSION_FILE environment variable is not set.")
+else()
+    include(${CCCL_VERSION_FILE})
+    if(DEFINED CCCL_VERSION_MAJOR AND DEFINED CCCL_VERSION_MINOR AND DEFINED CCCL_VERSION_PATCH)
+        message("${CCCL_VERSION_MAJOR}.${CCCL_VERSION_MINOR}.${CCCL_VERSION_PATCH}")
```

I did some testing and found that this actually prints to `stderr`, not `stdout` - my bad, I should have realized that sooner (though I don't think I've ever personally used the untyped `message()` signature.) Here's what you want:

```suggestion
        execute_process(COMMAND "${CMAKE_COMMAND}" -E echo "${CCCL_VERSION_MAJOR}.${CCCL_VERSION_MINOR}.${CCCL_VERSION_PATCH}")
```

---

I worked around this using `2>&1`. I can change this if you feel strongly but I kind of preferred piping both together since it’s calling cmake in a subshell.

---

IMO, `2>&1` is for when you just want to pipe the output of a long command into a human-readable log file. When a program has machine-readable output, `stdout` is meant for said output, and `stderr` is meant for human-readable error messages. Piping them together muddies this distinction and potentially injects error messages into places they shouldn't go. I'd rather spawn `cmake -E echo` in a subshell.

---

out of interest, why not use `message(STATUS ...)`? it's the only thing that goes to stdout and the FATAL_ERRORs go to stderr. Why's a further subprocess necessary?

---

then remove the `2>&1`

---

`message(STATUS ...)` prints an extra `-- ` at the beginning, which we're trying to avoid (though I suppose we could just filter it out on the bash side.)